modules/lang-painless/src/main/java/org/elasticsearch/painless/node/SSource.java

```diff
@@ -324,6 +315,38 @@ void write(MethodWriter writer, Globals globals) {
             writer.visitInsn(Opcodes.ACONST_NULL);
             writer.returnValue();
         }
+
+        writer.mark(endTry);
```

Comment:
I did move it out for the reason you mentioned. The error messages become very difficult to make sense of without the exception interpretation. I agree that it is unfortunate that this references a class in Elasticsearch core, but I think that is a thing we can fix later.

The other thing I like about this is that it makes the internals required to generate the right exception not publicly facing.

I'm happy to back it out propose it separately. Without it the exceptions that don't go through ScriptImpl are wonky though.